### PR TITLE
Add post processing flow

### LINKED ISSUE #6: Add post processing lambda with SES
Add a lambda function to update the database upon trigger. It should also additionally send out an email via SES if the created job had requested for a post job notification.

### PR DESCRIPTION
**Pull request by bearpolar**
Resolves #6 

Postprocessing lambda with SES

### CODE REVIEW

File: src/api/update-database.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+
+exports.updateDatabaseHandler = async (event) => {
+  
+  // All log statements are written to CloudWatch
+  console.info('received:', event);
+
+  // Get the message from the queue
+  sqs.receiveMessage(queue_params, function(err, data) {
```
Comment (madanalogy) [inline src/api/update-database.js:36]: This implementation pulls the messages from the SQS queue directly. What you need to do is read the message from the event source

File: src/api/send-email.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+
+exports.sendEmailHandler = async (event) => {
+  
+  // All log statements are written to CloudWatch
+  console.info('received:', event);
+
+  // Get the message from the queue
+  sqs.receiveMessage(queue_params, function(err, data) {
```
Comment (madanalogy) [inline src/api/send-email.js:35]: Similarly here

File: template.yaml [hunk 3]
```diff
@@ -111,6 +111,50 @@ Resources:
...
+      Environment:
+        Variables:
+          # Make queue URL accessible as environment variable from function code during execution
+          EMAIL_QUEUE: !Ref EmailQueue
+      Events:
+        SendEmailQueueEvent:
+          Type: SQS
+          Properties:
+            Queue: !GetAtt EmailQueue.Arn
+            BatchSize: 10
```
Comment (madanalogy) [inline template.yaml:147]: You can see here that the queue is specified as the event source, so the messages should come from the lambda `event` parameter.

**Review by madanalogy — CHANGES_REQUESTED** [review]
Template looks good! Just the function handlers that might need some tweaking. You can take a look at these resources for reference:

- https://docs.aws.amazon.com/lambda/latest/dg/with-sqs-example.html
- https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html

File: src/api/update-database.js (hunk 1 above)
Comment (bearpolar) [reply src/api/update-database.js]: replaced sqs.receiveMessage function with SQS event source mapping for Lambda

File: template.yaml (hunk 3 above)
Comment (madanalogy) [inline template.yaml:147]: Actually, maybe have the function process one message at a time, so `BatchSize: 1` haha

File: src/api/send-email.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  // Process each record in the queue and get the message body
+  // https://docs.aws.amazon.com/lambda/latest/dg/with-sqs-example.html
+  // https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html
+  try {
+    event.Records.forEach(record => {
+      const { body } = record;
+      console.log(body);
+    });
+  } catch (err) { console.log("Receive Error", err); }
```
[the remark this answers is not in the document]
Comment (bearpolar) [reply src/api/send-email.js]: updated as suggested

File: template.yaml (hunk 3 above)
Comment (bearpolar) [reply template.yaml]: updated BatchSize: 1

**Review by madanalogy — APPROVED** [review]
LGTM

**Pull request merged**